apis/visibility/v1alpha1/types.go

```diff
@@ -27,6 +27,11 @@ import (
 type PendingWorkload struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
They correspond to the workload object, similarly as for the visibility in status we decided. Otherwise we also need to return the "kind". Keeping the objects small was the motivation for the status visibility, here I think we can afford more fields easily. So we could have a follow up with JobName, and jobKind fields.

Reply:
I think the other way makes more sense, since you are already adding TypeMeta here.

Reply:
I see, I think we could try that, but indeed I suggest a follow up, maybe we should update the KEP for it.

I thought the `TypeMeta` here is to indicate details about the `PendingWorkload` type itself, isn't it?

Reply:
Right now we don't have an endpoint for an individual Workload or Job, so we are not actually using the TypeMeta right now.
But unclear if we will need a different struct for that.

Reply:
We intend to use the same struct `PendingWorkload` for all three endpoints: clusterqueue, localqueue, and individual workload. 

Just to be clear on the suggestion. `metav1.TypeMeta` has `Kind` and `APIVersion` fields. So, say, for a batch job we would return instance of `PendingWorkload` with `.kind=Job`, `.apiversion=batch/v1`, then `.name={job.name}`. 

I like the suggestion, so that we use the concept of Jobs, which are supplied by users. However, as you mention let's keep it as a follow up. We may also have a small KEP update to clarify this.

Reply:
To maintain this discussion I'm mentioning [issue#168](https://github.com/kubernetes-sigs/kueue/issues/168) so we do not lose visibility of it.